src/sync_manager.hpp

```diff
@@ -58,39 +66,32 @@ friend struct SyncSession;
     void set_log_level(util::Logger::Level) noexcept;
     void set_logger_factory(SyncLoggerFactory&) noexcept;
     void set_error_handler(std::function<sync::Client::ErrorHandler>);
-    void set_login_function(SyncLoginFunction);
+    void set_bind_session_function(SyncBindSessionFunction);
 
     /// Control whether the sync client attempts to reconnect immediately. Only set this to `true` for testing purposes.
     void set_client_should_reconnect_immediately(bool reconnect_immediately);
     /// Control whether the sync client validates SSL certificates. Should *always* be `true` in production use.
     void set_client_should_validate_ssl(bool validate_ssl);
 
-    std::shared_ptr<SyncSession> get_session(const std::string& path, const SyncConfig& config);
-    std::shared_ptr<SyncSession> get_existing_active_session(const std::string& path) const;
+    // Get a sync user for a given identity, or create one if none exists yet, and set its token.
+    // If a logged-out user exists, it will marked as logged back in.
+    shared_ptr<SyncUser> get_user(std::string identity, std::string user_token, bool is_admin=false);
```

Comment:
I agree. Mentioning systems that have yet to be designed doesn't help with anything and just confuses people. I'll fix that when I revise the PR later.

As for your second question, I think what you mean (please correct if I misunderstood) is that with the current system a session can be created for a logged-out user and only bound later (once the user is logged in), but with this system the session would need the user's refresh token at creation time (because of the callback). This is a good point.

I think the solution, with the proposed design, is no longer to have `SyncSession`s automatically start trying to get their access token as soon as they're created. Instead, when vended out from the user, the user (upon creating a session) can look at its current state and decide whether to ask the session to connect immediately, or to hold off until it's been logged in. This is pretty much what the binding does right now: a `RLMUser` will either make the network call to bind a session immediately, or it'll save the data necessary to do so and make that network call only once logged back in.